R/github_token.R

```diff
@@ -294,3 +296,48 @@ scold_for_scopes <- function(scopes) {
   )
   ui_bullets(message)
 }
+
+#' Lock and unlock a branch on GitHub
+#'
+#' These functions lock and unlock a branch on GitHub so that it's not possible
+#' for anyone to make any changes. This is used as part of the release process
+#' to ensure that you don't accidentally merge any pull requests or push any
+#' commits while you are waiting for CRAN to get back to you.
+#'
+#' @export
+#' @param branch The branch to lock/unlock. If not supplied, uses the
+#'   default branch with is usually "main" or "master".
+gh_lock_branch <- function(branch = NULL) {
+  cfg <- github_remote_config(github_get = TRUE)
+  repo <- target_repo(cfg)
+  branch <- branch %||% git_default_branch_(cfg)
+
+  invisible(gh::gh(
```

Comment:
What happens if you lock an already locked branch?

Reply:
Looks like it's idempotent